### PR TITLE
N-10 Refresh token

### PR DESCRIPTION
**Pull request by Zeroya**

### CODE REVIEW

File: backend/middlewares/checkAuth.js [hunk 1]
```diff
@@ -4,18 +4,20 @@ import * as dotenv from "dotenv";
...
-  jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, decoded) => {
-    if (err) {
-      return next(createError({ status: 401, message: "Unauthorized, invalid token" }));
-    }
-    req.user = decoded;
-    return next();
-  });
+  try {
+    const user = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET);
+    console.log(req.user);
```
Comment (Yuliia143) [inline backend/middlewares/checkAuth.js]: Remove, please

File: backend/middlewares/checkAuth.js [hunk 2]
```diff
@@ -4,18 +4,20 @@ import * as dotenv from "dotenv";
...
+    req.user = user;
+    console.log(req.user);
```
Comment (Yuliia143) [inline backend/middlewares/checkAuth.js]: x2

File: backend/controllers/authController.js [hunk 3]
```diff
@@ -21,14 +23,29 @@ const authLogin = async (req, res) => {
       expiresIn: "10s",
     });
 
+    const refreshToken = jwt.sign({ userId: user._id }, process.env.REFRESH_TOKEN_SECRET, {
+      expiresIn: "1d",
+    });
+
     res.cookie("jwt", token, {
+      maxAge: 10 * 10 * 100,
```
Comment (Yuliia143) [inline backend/controllers/authController.js]: Magic numbers should be constants

File: backend/controllers/authController.js [hunk 4]
```diff
@@ -21,14 +23,29 @@ const authLogin = async (req, res) => {
...
+      httpOnly: true,
+      secure: true,
+      sameSite: "none",
+    });
+
+    res.cookie("refresh", refreshToken, {
+      maxAge: 60 * 60 * 24 * 1000,
```
Comment (Yuliia143) [inline backend/controllers/authController.js]: x2

File: backend/controllers/authController.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import Users from "../models/userModal.js";
-import bcrypt from "bcryptjs";
+// import bcrypt from "bcryptjs";
```
Comment (Yuliia143) [inline backend/controllers/authController.js:2]: Do you need this?

File: backend/controllers/authController.js [hunk 6]
```diff
@@ -39,25 +56,57 @@ const authLogin = async (req, res) => {
...
+  jwt.verify(refreshToken, process.env.REFRESH_TOKEN_SECRET, (err, user) => {
+    err && console.log(err);
+    const { userId } = user;
+
+    const token = jwt.sign({ userId: userId }, process.env.ACCESS_TOKEN_SECRET, {
+      expiresIn: "10s",
+    });
+
+    res.cookie("jwt", token, {
+      maxAge: 10 * 10 * 100,
```
Comment (Yuliia143) [inline backend/controllers/authController.js]: x3

File: backend/controllers/todosController.js [hunk 7]
```diff
@@ -61,19 +51,48 @@ const updateTodo = async (req, res) => {
...
     const deleteItem = await Todos.findByIdAndDelete(req.params.id);
-    res.status(200).json(deleteItem);
+    return res.status(200).json(deleteItem);
   } catch (error) {
     res.status(500).json(error);
   }
 };
 
-export { getTodos, addTodo, toggleTodoDone, updateTodo, deleteTodo };
+const buttonFilterTodos = async (req, res) => {
```
Comment (Yuliia143) [inline backend/controllers/todosController.js]: Is it a good name for request - 'button' ?

File: frontend/src/components/Auth/Login.tsx [hunk 8]
```diff
@@ -34,7 +34,9 @@ const Login: FC = () => {
         return;
       }
       setChecker(true);
-      await loginUser(form).then((response) => getToken(response.data.token, response.data.user.userId));
+      await loginUser(form).then((response) =>
+        getToken(response.data.token, response.data.refreshToken, response.data.user.userId)
```
Comment (Yuliia143) [inline frontend/src/components/Auth/Login.tsx]: Use destructuring

File: frontend/src/hooks/auth.ts [hunk 9]
```diff
@@ -9,31 +9,33 @@ export const useAuth = () => {
   const [userId, setUserId] = useState("");
   const [isReady, setIsReady] = useState(false);
 
-  const login = useCallback((jwtToken: any, id: string) => {
+  const login = useCallback((jwtToken: any, refreshToken: string, id: string) => {
```
Comment (Yuliia143) [inline frontend/src/hooks/auth.ts]: Why any?

File: backend/controllers/authController.js (hunk 5 above)
Comment (Zeroya) [reply backend/controllers/authController.js:2]: I will need it in the sign up task, therefore I did not delete it

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**